### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by jamesplease** [rollup/rollup#2085]
This is similar to a slew of other issues reported here about max call stack sizes. The steps to reproduce are:

1. clone [React Composer](https://github.com/jamesplease/react-composer)
2. `git reset --hard 14bff07d64`
3. `npm run build:umd:min`
4. observe the error

The relevant Babel config is:

```json
{
  "presets": [["env", { "modules": false }], "stage-3", "react"],
  "plugins": [
    "external-helpers",
    "transform-class-properties",
    [
      "transform-react-remove-prop-types",
      {
        "mode": "remove",
        "removeImport": true
      }
    ]
  ]
}
```

The Rollup config is:

```js
import nodeResolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import babel from 'rollup-plugin-babel';
import uglify from 'rollup-plugin-uglify';
import replace from 'rollup-plugin-replace';

var env = process.env.NODE_ENV;
var config = {
  output: {
    name: 'ReactComposer',
    globals: {
      react: 'React'
    },
    format: 'umd'
  },
  external: ['react'],
  context: 'this',
  plugins: [
    nodeResolve({
      jsnext: true
    }),
    commonjs({
      include: 'node_modules/**'
    }),
    babel({
      exclude: 'node_modules/**'
    }),
    replace({
      'process.env.NODE_ENV': JSON.stringify(env)
    })
  ]
};

if (env === 'production') {
  config.plugins.push(
    uglify({
      compress: {
        pure_getters: true,
        unsafe: true,
        unsafe_comps: true,
        warnings: false
      }
    })
  );
}

export default config;
```

This was also happening on Rollup ~v0.45. I just updated and am still getting the error. The stack trace is:

```
src/index.js → dist/react-composer.min.js...
[!] (babel plugin) RangeError: /Users/jmeas/webdev/react-projects/react-composer/src/index.js: Maximum call stack size exceeded
src/index.js
RangeError: /Users/jmeas/webdev/react-projects/react-composer/src/index.js: Maximum call stack size exceeded
    at error (/Users/jmeas/webdev/react-projects/react-composer/node_modules/rollup/dist/rollup.js:224:15)
    at Object.error (/Users/jmeas/webdev/react-projects/react-composer/node_modules/rollup/dist/rollup.js:17174:21)
    at /Users/jmeas/webdev/react-projects/react-composer/node_modules/rollup/dist/rollup.js:17183:29
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:686:11)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```

Let me know if you need any more information. Thanks!

**Comment by lukastaegert** [comment]
Reading the error message it seems that in contrast to other issues, this happens inside the babel-plugin and should probably reported there: https://github.com/rollup/rollup-plugin-babel/issues

**Comment by jamesplease** [comment]
Sounds good. I’ll move this on over there, then. Thanks @lukastaegert !

**Comment by jamesplease** [comment]
Just a quick update for any future developer visitors: this was ultimately a problem in `babel-plugin-transform-react-remove-prop-types`, and it has been fixed as of `v0.4.14` of that lib.
